From b45a17f56211de7156a037d431a188ad853f21ca Mon Sep 17 00:00:00 2001
From: Richard Goodman <richardg@brandwatch.com>
Date: Wed, 20 May 2020 12:51:43 +0100
Subject: [PATCH 35/83] Force ant to build every time

---
 Jenkinsfile | 36 ++++++------------------------------
 1 file changed, 6 insertions(+), 30 deletions(-)

diff --git a/Jenkinsfile b/Jenkinsfile
index 9629704259..7247e4a48f 100644
--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -1,24 +1,5 @@
 #!groovy
 gcpProject = "bw-prod-platform0"
-baseImageExists = false
-
-def doesBaseImageExist(String imageVersion) {
-    /*
-        Originally, the docker file would create a new debian image and install
-        all the dependencies needed to build solr, to save time, I've separated
-        the two "stages" into separate docker files, with this, I then do a check
-        to see if the ant-base image exists in our gcr, if it does, then it'll skip
-        building it which should cut down on processing time
-    */
-    try {
-        img = docker.image("${gcpProject}/ant-base:${imageVersion}").withRun { c ->
-            sh 'ant --help'
-        }
-        return true
-    } catch (Exception e) {
-        return false
-    }
-}
 
 def pushImage(String imageID, String imageVersion) {
     buildNumber = "build_${env.BUILD_NUMBER}"
@@ -47,18 +28,13 @@ node('docker') {
     }
 
     stage ('Build images') {
-        baseImageExists = doesBaseImageExist(ant_version)
-        if ( !baseImageExists ) {
-            echo "Unable to find base image, building"
-            withEnv(["DOCKER_BUILDKIT=1"]) {
-                sh """ docker build -f base.Dockerfile \
-                    --build-arg ANT_VERSION=${ant_version} \
-                    -t ${gcpProject}/ant-base:${ant_version} .
-                """
-                pushImage("${gcpProject}/ant-base:${ant_version}", ant_version)
-            }
-        }
         withEnv(["DOCKER_BUILDKIT=1"]) {
+            sh """ docker build -f base.Dockerfile \
+                --build-arg ANT_VERSION=${ant_version} \
+                -t ${gcpProject}/ant-base:${ant_version} .
+            """
+            pushImage("${gcpProject}/ant-base:${ant_version}", ant_version)
+
             sh """ docker build \
                 --build-arg REPO=${gcpProject} \
                 --build-arg ANT_VERSION=${ant_version} \
-- 
2.32.1 (Apple Git-133)

